Marlin/temperature.cpp

```diff
@@ -203,7 +203,7 @@ uint8_t Temperature::soft_pwm[HOTENDS];
   int Temperature::current_raw_filwidth = 0;  //Holds measured filament diameter - one extruder only
 #endif
 
-#ifdef ADC_KEYPAD
+#if ENABLED(ADC_KEYPAD) && PIN_EXISTS(ADC_KEYPAD)
```

Comment:
I changed stuff in the "base" already. Without having an ADC_KEYPAD (which is the stock Anet A8 Keypad) you can't test any of the changes (esp. see stuff below)